unittests/InstructionCountCI/FlagM/HotBlocks.json

```diff
@@ -0,0 +1,40 @@
+{
+  "Features": {
+    "Bitness": 64,
+    "EnabledHostFeatures": [
+      "FLAGM",
+      "FLAGM2"
+    ],
+    "DisabledHostFeatures": [
+      "SVE128",
+      "SVE256"
+    ]
+  },
+  "Instructions": {
+    "The Witcher 3": {
+      "ExpectedInstructionCount": 11,
+      "x86Insts": [
+        "mov eax, 0x1",
+        "lock xadd qword [rcx], rax",
+        "mov rdx, rax",
+        "and edx, 0x1f",
+        "inc rdx",
+        "shl rdx, 0x6",
+        "add rdx, rcx"
+      ],
+      "ExpectedArm64ASM": [
+        "mov w4, #0x1",
+        "ldaddal x4, x4, [x5]",
+        "mov x6, x4",
```

Comment:
We should really be able to DCE this move, I guess SRA is getting in the way of regular DCE doing it, so it'd need to be either RCLSE or something like the flag elim pass. @Sonicadvance1 thoughts which is more appropriate?

Reply:
Not sure, ldaddal will update the value in rax but then it also copies the value in to rdx there. Can it actually be eliminated unless we know this is a looping block or if rax later gets overwritten?

Reply:
yes, x6 is overwritten later in the block

Reply:
Tracking in #3444

Reply:
x6 is overwritten yes, which corresponds to rdx. But x4 which corresponds to rax is not. So rax and rdx contain the same returned value from ldaddal once the move occurs

Reply:
..so? The sequence is

        "mov x6, x4",
        "and w6, w4, #0x1f",

The mov can be deleted without affecting either register after the and.

Reply:
Oop, yep. I was on my phone so I didn't capture the full context that rdx is overwritten so it can remove the mov in favour of using the ALU operation on rax directly.